pincer/objects/guild/guild.py

```diff
@@ -1916,6 +1919,266 @@ async def get_webhooks(self) -> AsyncGenerator[Webhook, None]:
+        Returns an async generator of the guild scheduled events.
+
+        Parameters
+        ----------
+        with_user_count : :class:`bool`
+            Whether to include the user count in the scheduled event.
+
+        Yields
+        ------
+        :class:`~pincer.objects.guild.scheduled_event.ScheduledEvent`
+            The scheduled event object.
+        """
+        data = await self._http.get(
+            f"guilds/{self.id}/scheduled-events?{with_user_count=!s}"
+        )
+        for event_data in data:
+            yield ScheduledEvent.from_dict(event_data)
+
+    async def create_scheduled_event(
+        self,
+        name: str,
+        privacy_level: int,
+        entity_type: int,
+        scheduled_start_time: datetime,
+        scheduled_end_time: Optional[datetime] = None,
+        entity_metadata: Optional[str] = None,
+        channel_id: Optional[int] = None,
+        description: Optional[str] = None,
+        reason: Optional[str] = None,
+    ) -> ScheduledEvent:
+        """
+        Create a new scheduled event for the guild.
+
+        Parameters
+        ----------
+        name : :class:`str`
+            The name of the scheduled event.
+        privacy_level : :class:`int`
+            The privacy level of the scheduled event.
+        entity_type : :class:`int`
+            The type of entity to be scheduled.
+        scheduled_start_time : :class:`datetime`
+            The scheduled start time of the event.
+        scheduled_end_time : Optional[:class:`datetime`]
+            The scheduled end time of the event.
+        entity_metadata : Optional[:class:`str`]
+            The metadata of the entity to be scheduled.
+        channel_id : Optional[:class:`int`]
+            The channel id of the channel to be scheduled.
+        description : Optional[:class:`str`]
+            The description of the scheduled event.
+        reason : Optional[:class:`str`]
+            The reason for creating the scheduled event.
+
+        Returns
+        -------
+        :class:`~pincer.objects.guild.scheduled_event.ScheduledEvent`
+            The newly created scheduled event.
+        """
+        if scheduled_start_time.timestamp() < datetime.now().timestamp():
+            raise ValueError("An event cannot be created in the past")
+
+        if (
+            scheduled_end_time is not None
+            and scheduled_end_time.timestamp() < datetime.now().timestamp()
+        ):
+            raise ValueError("An event cannot start before it ends")
+
+        data = await self._http.post(
+            f"guilds/{self.id}/scheduled-events",
+            data={
+                "name": name,
+                "privacy_level": privacy_level,
+                "entity_type": entity_type,
+                "scheduled_start_time": scheduled_start_time.isoformat(),
+                "scheduled_end_time": scheduled_end_time.isoformat()
+                if scheduled_end_time is not None
+                else None,
```

Comment:
:thinking: :thinking: :thinking: 
```suggestion
                "scheduled_end_time": getattr(
                    scheduled_end_time, "isoformat", type(None)
                )(),
```